core/src/test/scala/com/gu/tableversions/core/TableVersionsSpec.scala

```diff
@@ -0,0 +1,194 @@
+package com.gu.tableversions.core
+
+import java.time.Instant
+
+import cats.effect.IO
+import com.gu.tableversions.core.Partition.PartitionColumn
+import com.gu.tableversions.core.TableVersions.CommitResult.SuccessfulCommit
+import com.gu.tableversions.core.TableVersions._
+import org.scalatest.{FlatSpec, Matchers}
+
+/**
+  * Spec containing tests that apply across all TableVersions implementations.
+  *
+  * These are black box tests purely in terms of the TableVersions interface.
+  */
+trait TableVersionsSpec {
+  this: FlatSpec with Matchers =>
+
+  def tableVersionsBehaviour(emptyTableVersions: IO[TableVersions[IO]]): Unit = {
+
+    val table = TableName("schema", "table")
+    val userId = UserId("Test user")
+    val date = PartitionColumn("date")
+
+    it should "have an idempotent 'init' operation" in {
```

Comment:
Should this assert against the result of the first `.init` *and* subsequent ones?